energy-integration/governance-v2/src/lib.rs

```diff
@@ -153,110 +153,42 @@ pub trait GovernanceV2:
+        let voting_power = user_energy.sqrt();
 
         match vote {
             VoteType::UpVote => {
                 self.proposal_votes(proposal_id).update(|proposal_votes| {
-                    proposal_votes.up_votes += &user_energy.clone();
+                    proposal_votes.up_votes += &voting_power.clone();
                 });
-                self.up_vote_cast_event(&voter, proposal_id, &user_energy);
+                self.up_vote_cast_event(&voter, proposal_id, &voting_power);
             }
             VoteType::DownVote => {
                 self.proposal_votes(proposal_id).update(|proposal_votes| {
-                    proposal_votes.down_votes += &user_energy.clone();
+                    proposal_votes.down_votes += &voting_power.clone();
                 });
-                self.down_vote_cast_event(&voter, proposal_id, &user_energy);
+                self.down_vote_cast_event(&voter, proposal_id, &voting_power);
             }
             VoteType::DownVetoVote => {
                 self.proposal_votes(proposal_id).update(|proposal_votes| {
-                    proposal_votes.down_veto_votes += &user_energy.clone();
+                    proposal_votes.down_veto_votes += &voting_power.clone();
                 });
-                self.down_veto_vote_cast_event(&voter, proposal_id, &user_energy);
+                self.down_veto_vote_cast_event(&voter, proposal_id, &voting_power);
             }
             VoteType::AbstainVote => {
                 self.proposal_votes(proposal_id).update(|proposal_votes| {
-                    proposal_votes.abstain_votes += &user_energy.clone();
+                    proposal_votes.abstain_votes += &voting_power.clone();
                 });
-                self.abstain_vote_cast_event(&voter, proposal_id, &user_energy);
-            }
-        }
-    }
-
-    /// Queue a proposal for execution.
-    /// This can be done only if the proposal has reached the quorum.
-    /// A proposal is considered successful and ready for queing if
-    /// total_votes + total_downvotes >= quorum && total_votes > total_downvotes
-    /// i.e. at least 50% + 1 of the people voted "yes".
-    ///
-    /// Additionally, all the required payments must be deposited before calling this endpoint.
-    #[endpoint]
-    fn queue(&self, proposal_id: ProposalId) {
-        self.require_caller_not_self();
-        require!(
-            self.get_proposal_status(proposal_id) == GovernanceProposalStatus::Succeeded,
-            "Can only queue succeeded proposals"
-        );
-
-        let current_block = self.blockchain().get_block_nonce();
-        self.proposal_queue_block(proposal_id).set(current_block);
-
-        self.proposal_queued_event(proposal_id, current_block);
-    }
-
-    /// Execute a previously queued proposal.
-    /// This will also clear the proposal from storage.
-    #[endpoint]
-    fn execute(&self, proposal_id: ProposalId) {
```

Comment:
We no longer have the option to automatically execute proposal actions? If so, should we remove the GovernanceAction list from the GovernanceProposal (and all that comes with it -> like gas variable and checks).

Reply:
we might return these - but into a separate proxy contract - on v2.